reader.go

```diff
@@ -1483,6 +1483,37 @@ func (r *Reader) SetOffset(offset int64) error {
 	return err
 }
 
+// SetOffsetWithTimestamp changes the offset from which the next batch of messages will be
+// read given the timestamp t.
+//
+// The method fails if the unable to connect partition leader, or unable to read the offset
+// given the ts, or if the reader has been closed.
+func (r *Reader) SetOffsetWithTimestamp(ctx context.Context, t time.Time) error {
```

Comment:
```suggestion
func (r *Reader) SetOffsetAt(ctx context.Context, t time.Time) error {
```
I'm not particularly attached to this suggestion, I feel like the current name is a bit verbose and redundant tho.